packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -310,6 +319,18 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
      */
     protected abstract loadCore(services: IChannelStorageService): Promise<void>;
 
+    /**
+     * Called when a handle id decoded by this object. A handle in the object's data represents a reference to another
+     * object in the container.
+     * @param handle - The handle of the Fluid object that is decoded.
+     */
+    protected handleDecoded(handle: IFluidHandle) {
+        if (this.isAttached()) {
+            // The handle is decoded by this object itself. So, the id is "/" representing this object.
+            this.services?.deltaConnection.referenceAdded?.("/", handle);
```

Comment:
Is TSC not smart enough to know that this.services !== undefined here? Can we change signature of isAttached to help it?

Reply:
Nope, tsc cannot detect this. I don't think doing anything inside isAttached will help tsc here.

Reply:
I was thinking to use "asserts" keyword (same way as we use with assert() function), but I do not see an easy way to do it for class member function (referring to "this" object). Maybe it exists, I simply do not know how to make it happen